week-03/assessment/solar-farm/solar-farm/src/test/java/Domain/PanelServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package Domain;
+
+import Data.PanelRepositoryDouble;
+import learn.solar.data.DataException;
+import learn.solar.data.PanelRepository;
+import learn.solar.domain.PanelResult;
+import learn.solar.domain.PanelService;
+import learn.solar.models.Material;
+import learn.solar.models.Panel;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+//panelservice test
+public class PanelServiceTest {
+    PanelRepository repository = new PanelRepositoryDouble(); // we are using our double here
+    PanelService service = new PanelService(repository);
+
+    @Test
+    public void shouldCreatePage() throws DataException {
+        PanelResult actual = service.add(new Panel(8,"west", 1, 1, 2019, Material.AMORPHOUS_SILICON, false));
+
+        //verify that the payload is set - make sure its not null
+        assertNotNull(actual.getPanel());
+        assertTrue(actual.isSuccess());
+
```

Comment:
we will need to test all validation 
- missing duplicate (section, row, column)
- missing blank section 
- missing null